### PR TITLE
Update README.md for Fleet Server

### PR DESCRIPTION
**Pull request by mostlyjason**
Adding a description to address https://github.com/elastic/integrations/issues/787

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

## Author's Checklist

- [ ] I'm not a dev, so please check I'm not breaking anything 🤞 

## Related issues

- Closes https://github.com/elastic/integrations/issues/787

### CODE REVIEW

**Comment by ruflin** [comment]
You need to bump the version number in the manifest.yml and add a changelog entry to the changelog file so it gets released automatically aftewards to snapshot.

File: packages/fleet_server/docs/README.md [hunk 1]
```diff
@@ -1 +1,8 @@
-# Fleet Server package
\ No newline at end of file
+# Fleet Server integration
+
+Fleet Server is a component of the Elastic Stack used to centrally manage Elastic Agents. It’s launched as part of an Elastic Agent on a host intended to act as a server. One Fleet Server process can support many Elastic Agent connections. It is responsible for updating agent policies, collecting status information, and coordinating actions across Elastic Agents.
+
+To add this integration to an Elastic Agent, just add it to an agent policy. The Elastic Agents enrolled into that policy must run with additional credentials such as a service token. Learn how to add a Fleet Server our [documentation](https://www.elastic.co/guide/en/fleet/current/fleet-server.html).
+
+## Compatibility
```
Comment (ruflin) [inline packages/fleet_server/docs/README.md:7]: Should we update this in the future to mention more precise how the compatiblity is? fleet-server needs to newer or equal Elastic Agents and the same with ES?

**Review by ruflin — COMMENTED** [review]
Change LGTM as soon as changelog and version number is increased.

File: packages/fleet_server/docs/README.md (hunk 1 above)
Comment (ph) [reply packages/fleet_server/docs/README.md:7]: I think you mean `Elastic Agent version must match Elasticsearch version?`?

**Review by ph — APPROVED** [review]
@mostlyjason LGMT added a small notes about compatibility, but the text looks good to me and the version bump is also OK.

**Review by urso — APPROVED** [review]

File: packages/fleet_server/docs/README.md (hunk 1 above)
Comment (mostlyjason) [reply packages/fleet_server/docs/README.md:7]: @ph I think the Elasticsearch version is addressed by the sentence "Fleet Server is compatible with the Elastic Stack and Elastic Agents version 7.13 or higher"

**Pull request merged**